plugins/ScheduledReports/API.php

```diff
@@ -345,88 +359,98 @@ public function generateReport($idReport, $date, $language = false, $outputType
         $initialFilterTruncate = Common::getRequestVar('filter_truncate', false);
         $_GET['filter_truncate'] = Config::getInstance()->General['scheduled_reports_truncate'];
 
-        $prettyDate = null;
-        $processedReports = array();
-        $segment = self::getSegment($report['idsegment']);
-        foreach ($reportMetadata as $action) {
-            $apiModule = $action['module'];
-            $apiAction = $action['action'];
-            $apiParameters = array();
-            if (isset($action['parameters'])) {
-                $apiParameters = $action['parameters'];
-            }
-
-            $mustRestoreGET = false;
-
-            // all Websites dashboard should not be truncated in the report
-            if ($apiModule == 'MultiSites') {
-                $mustRestoreGET = $_GET;
-                $_GET['enhanced'] = true;
-
-                if ($apiAction == 'getAll') {
-                    $_GET['filter_truncate'] = false;
-                    $_GET['filter_limit'] = -1; // show all websites in all websites report
+        $originalShowEvolutionWithinSelectedPeriod = Config::getInstance()->General['graphs_show_evolution_within_selected_period'];
+        $originalDefaultEvolutionGraphLastPeriodsAmount = Config::getInstance()->General['graphs_default_evolution_graph_last_days_amount'];
+        try {
+            Config::getInstance()->General['graphs_show_evolution_within_selected_period'] = (bool)$report['evolution_graph_within_period'];
```

Comment:
I haven't tested it yet but are you sure this works? I think in the past usually it was often needed to do

`$general = $config->General;$general['...']=(bool)....; $config->General = $general;`. Not sure if this is still a thing...

Also I'm not sure this is actually applied... without testing or looking further I would have expected this needs to be set before calling `\Piwik\Plugins\API\API::getInstance()->getReportMetadata($idSite);`

Reply:
I remember that was for specific versions of PHP. I can do that to be safe.

Reply:
Note: it's used in `API.getProcessedReport`, but can set it before all the API calls.

Reply:
> I remember that was for specific versions of PHP. I can do that to be safe.

might be good. I don't really remember why it was needed or whether it's still the case. Haven't tested to write it the normal way in a while as gotten used to it :)